File: backend/hotelEase/hotelEase.Model/Room.cs
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Text;

namespace hotelEase.Model
{
    public class Room
    {
        public int Id { get; set; }

        public int HotelId { get; set; }

        public int RoomTypeId { get; set; }

        public string Name { get; set; } = null!;

        public int Capacity { get; set; }

        public decimal PricePerNight { get; set; }

        public bool? IsAvailable { get; set; }

        public string? Description { get; set; }
        public bool? QueenBed { get; set; }
        public bool? WiFi { get; set; }
        public bool? CityView { get; set; }
        public bool? AC { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? DeletedTime { get; set; }
        public virtual ICollection<Asset> Assets { get; set; } = new List<Asset>();
    }
}

File: UI/hotelease_mobile_new/lib/models/room.dart
import 'package:json_annotation/json_annotation.dart';
import 'asset.dart';

part 'room.g.dart';

@JsonSerializable()
class Room {
  int? id;
  int? hotelId;
  int? roomTypeId;
  String? name;
  int? capacity;
  double? pricePerNight;
  bool? isAvailable;
  String? description;
  List<Asset>? assets;

  bool? queenBed;
  bool? wiFi;
  bool? cityView;
  bool? ac;
  Room({
    this.id,
    this.hotelId,
    this.roomTypeId,
    this.name,
    this.capacity,
    this.pricePerNight,
    this.isAvailable,
    this.description,
    this.assets,
    this.ac,
    this.cityView,
    this.queenBed,
    this.wiFi,
  });

  factory Room.fromJson(Map<String, dynamic> json) => _$RoomFromJson(json);
  Map<String, dynamic> toJson() => _$RoomToJson(this);
}
